src/Offer/Offer.php

```diff
@@ -564,34 +532,46 @@ protected function applyOrganizerDeleted(AbstractOrganizerDeleted $organizerDele
      */
     protected function applyImagesImportedFromUDB2(AbstractImagesImportedFromUDB2 $imagesImportedFromUDB2)
     {
-        $this->applyImagesEvent($imagesImportedFromUDB2);
+        $this->applyUdb2ImagesEvent($imagesImportedFromUDB2);
     }
 
     /**
      * @param AbstractImagesUpdatedFromUDB2 $imagesUpdatedFromUDB2
      */
     protected function applyImagesUpdatedFromUDB2(AbstractImagesUpdatedFromUDB2 $imagesUpdatedFromUDB2)
     {
-        $this->applyImagesEvent($imagesUpdatedFromUDB2);
+        $this->applyUdb2ImagesEvent($imagesUpdatedFromUDB2);
     }
 
     /**
-     * Overwrites or resets the main image and all media objects.
+     * This indirect apply method can be called internally to deal with images coming from UDB2.
+     * Imports from UDB2 only contain the native Dutch content.
+     * @see https://github.com/cultuurnet/udb3-udb2-bridge/blob/db0a7ab2444f55bb3faae3d59b82b39aaeba253b/test/Media/ImageCollectionFactoryTest.php#L79-L103
+     * Because of this we have to make sure translated images are left in place.
      *
      * @param AbstractImagesEvent $imagesEvent
      */
-    protected function applyImagesEvent(AbstractImagesEvent $imagesEvent)
+    protected function applyUdb2ImagesEvent(AbstractImagesEvent $imagesEvent)
     {
-        $mainImage = $imagesEvent->getImages()->getMain();
-
-        $this->mainImageId = $mainImage ? $mainImage->getMediaObjectId() : null;
+        $newMainImage = $imagesEvent->getImages()->getMain();
 
-        $this->mediaObjects = array_map(
+        $dutchImagesList = array_filter(
+            $imagesEvent->getImages()->toArray(),
             function (Image $image) {
-                return $image->getMediaObjectId();
-            },
-            $imagesEvent->getImages()->toArray()
+                return $image->getLanguage()->getCode() === 'nl';
```

Comment:
Het filteren van de inkomende afbeeldingen op taal is overbodig omdat deze toch altijd in het Nederlands zijn. De eerste filter in de offer aggregate is redundant. Logica afsplitsen is lastig omdat je op beide plaatsen vertrekt van een ander formaat. `ImageCollection` in offer VS. `json-ld` in projector.